subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/FileSystemWatchingBuildActionRunner.java

```diff
@@ -114,6 +114,11 @@ public boolean isEnabled() {
                 return actuallyWatching;
             }
         });
+        if (action.getStartParameter().isContinuous()) {
+            if (!actuallyWatching) {
+                throw new IllegalStateException("Continuous build requires file system watching to be enabled");
```

Comment:
I think a better message would be:
```suggestion
                throw new IllegalStateException("Continuous build does not work when file system watching is disabled");
```

...given that you have to explicitly disable FSW, no?

Reply:
We also need a test for this message.

Reply:
I added a test.